--- isdn4k-utils-CVS-2010-05-01-patched/isdnlog/tools/isdnrate.man.than	2016-05-02 23:04:28.963373668 +0200
+++ isdn4k-utils-CVS-2010-05-01-patched/isdnlog/tools/isdnrate.man	2016-05-02 23:08:54.720296768 +0200
@@ -3,7 +3,7 @@
 .SH NAME
 isdnrate \- Print telephone rates and various info from rate\-files(5).
 .SH SYNOPSIS
-.BI "isdnrate \-b" best " \-d" day " \-f" from " \-h" hour " \-l" len
+.BI "isdnrate \-a \-b" best " \-d" day " \-f" from " \-h" hour " \-l" len
 .B \-o
 .BI "\-p" provs " \-r"
 .I \fPv\fINN
@@ -57,6 +57,9 @@
 .SS
 .B OPTIONS
 .P
+.BI \-a
+Show old and newer rates. Default actual only.
+.P
 .BI \-b best
 Print only \fIbest\fP providers. Default is all.
 .P
