diff -up isdn4k-utils-CVS-2010-05-01-patched/doc/Configure.help isdn4k-utils-CVS-2010-05-01-patched/doc/Configure
diff -up isdn4k-utils-CVS-2010-05-01-patched/isdnlog/isdnrep/isdnbill.c.help isdn4k-utils-CVS-2010-05-01-patched/isdnlog/isdnrep/isdnbill.c
--- isdn4k-utils-CVS-2010-05-01-patched/isdnlog/isdnrep/isdnbill.c.help	2009-10-20 11:07:28.000000000 +0200
+++ isdn4k-utils-CVS-2010-05-01-patched/isdnlog/isdnrep/isdnbill.c	2016-05-09 12:24:55.180206330 +0200
@@ -139,8 +139,6 @@ typedef struct {
 } PARTNER;
 
 static char  options[]   = "nv:VioeaN:mftIE";
-static char  usage[]     = "%s: usage: %s [ -%s ]\n";
-
 
 static CALLER   c;
 static MSNSUM   msnsum[2][MAXSI][MAXMYMSN];
@@ -170,6 +168,36 @@ static char     onlythis[32] = { 0 }; /*
 static int	onlyInternal = 0;     /* -I    -> nur Verbindungen am Internen S0-Bus anzeigen */
 static int	onlyExternal = 0;     /* -E    -> nur Verbindungen am Externen S0-Bus anzeigen */
 
+static void isdnbill_usage() {
+      //printf("This makes no sense! You must specify -i, -o or -e\n");
+      printf("\t-a    -> alle Verbindungen anzeigen i.e. \"-ioe\"\n");
+      printf("\t      -> show all connections\n");
+      printf("\t-e    -> nichtzustandegekommene Verbindungen anzeigen\n");
+      printf("\t      -> show incomplete calls\n");
+      printf("\t-f    -> Verbindungsentgeld _immer_ neu berechnen\n");
+      printf("\t      -> recalculate costs _always_\n");
+      printf("\t-i    -> reinkommende Verbindungen anzeigen\n");
+      printf("\t      -> show incoming calls\n");
+      printf("\t-m    -> ohne MwSt anzeigen\n");
+      printf("\t      -> prices without VAT\n");
+      printf("\t-n    -> _nicht_ anstelle Rufnummern Alias-Bezeichnungen anzeigen\n");
+      printf("\t      -> do _not_ replace numbers with aliases\n");
+      printf("\t-o    -> rausgehende Verbindungen anzeigen\n");
+      printf("\t      -> show outgoing calls\n");
+      printf("\t-t    -> nur die heutigen Verbindungen anzeigen\n");
+      printf("\t      -> show only current connections\n");
+      printf("\t-vn   -> Verbose Level\n");
+      printf("\t-Nnnn -> nur Verbindungen mit _dieser_ Rufnummer anzeigen\n");
+      printf("\t      -> show only calls with the specified number\n");
+      printf("\t-I    -> nur Verbindungen am Internen S0-Bus anzeigen\n");
+      printf("\t      -> show only calls on the internal S0 bus\n");
+      printf("\t-E    -> nur Verbindungen am Externen S0-Bus anzeigen\n");
+      printf("\t      -> show only calls on the external S0 bus\n");
+      printf("\t-V    -> Version anzeigen\n");
+      printf("\t      -> show version\n");
+
+      exit(255);
+}
 
 int print_msg(int Level, const char *fmt, ...)
 {
@@ -910,39 +936,17 @@ int main(int argc, char *argv[], char *e
         case 'E' : onlyExternal++;
              	   break;
 
-        case '?' : printf(usage, argv[0], argv[0], options);
+        case '?' : isdnbill_usage();
                    return(1);
+
+        case 'h' :
+        default:
+                   isdnbill_usage();
+                   break;
       } /* switch */
 
     if (!showincoming && !showoutgoing && !showerrors) {
-      printf("This makes no sense! You must specify -i, -o or -e\n");
-      printf("\t-a    -> alle Verbindungen anzeigen i.e. \"-ioe\"\n");
-      printf("\t      -> show all connections\n");
-      printf("\t-e    -> nichtzustandegekommene Verbindungen anzeigen\n");
-      printf("\t      -> show incomplete calls\n");
-      printf("\t-f    -> Verbindungsentgeld _immer_ neu berechnen\n");
-      printf("\t      -> recalculate costs _always_\n");
-      printf("\t-i    -> reinkommende Verbindungen anzeigen\n");
-      printf("\t      -> show incoming calls\n");
-      printf("\t-m    -> ohne MwSt anzeigen\n");
-      printf("\t      -> prices without VAT\n");
-      printf("\t-n    -> _nicht_ anstelle Rufnummern Alias-Bezeichnungen anzeigen\n");
-      printf("\t      -> do _not_ replace numbers with aliases\n");
-      printf("\t-o    -> rausgehende Verbindungen anzeigen\n");
-      printf("\t      -> show outgoing calls\n");
-      printf("\t-t    -> nur die heutigen Verbindungen anzeigen\n");
-      printf("\t      -> show only current connections\n");
-      printf("\t-vn   -> Verbose Level\n");
-      printf("\t-Nnnn -> nur Verbindungen mit _dieser_ Rufnummer anzeigen\n");
-      printf("\t      -> show only calls with the specified number\n");
-      printf("\t-I    -> nur Verbindungen am Internen S0-Bus anzeigen\n");
-      printf("\t      -> show only calls on the internal S0 bus\n");
-      printf("\t-E    -> nur Verbindungen am Externen S0-Bus anzeigen\n");
-      printf("\t      -> show only calls on the external S0 bus\n");
-      printf("\t-V    -> Version anzeigen\n");
-      printf("\t      -> show version\n");
-
-      return(1);
+      isdnbill_usage();
     } /* if */
 
     *home = 0;
@@ -1320,11 +1324,14 @@ int main(int argc, char *argv[], char *e
       showpartner();
 
     }
-    else
+    else {
       fprintf(stderr, "%s: Can't read configuration file(s)\n", myname);
+      isdnbill_usage();
+    }
   }
-  else
+  else {
     fprintf(stderr, "%s: Can't open \"isdn.log\" file\n", myname);
-
+    isdnbill_usage();
+  }
   return(0);
 } /* isdnbill */
