--- isdn4k-utils-CVS-2010-05-01-patched/isdnlog/isdnlog/isdnlog.8.in.than	2016-05-02 23:57:40.348338954 +0200
+++ isdn4k-utils-CVS-2010-05-01-patched/isdnlog/isdnlog/isdnlog.8.in	2016-05-03 00:21:56.887073658 +0200
@@ -119,6 +119,18 @@
 displayed.
 
 .TP
+.B \-Hx  watchdog
+
+.TP
+.B \-N   use new config
+
+.TP
+.B \-Q   sqldump
+
+.TP
+.B \-a   any
+
+.TP
 .B \-u\fIx\fB	ignoreRR="value"
 Ignore "housekeeping" frames.
 .I x
