--- isdn4k-utils-CVS-2010-05-01-patched/isdnlog/tools/zone/mkzonedb.c.than	2016-05-02 17:54:16.761387266 +0200
+++ isdn4k-utils-CVS-2010-05-01-patched/isdnlog/tools/zone/mkzonedb.c	2016-05-02 18:13:28.749820644 +0200
@@ -65,8 +65,15 @@
 
 
 void usage(char *argv[]) {
-	fprintf(stderr, "%s: -r RedZonefile -d Database [ -v ] [ -V ] [ -o Localzone ] [ -l Len ]\n",
-		basename(argv[0]));
+   fprintf(stderr, "\nUsage: %s -r zonefile -d database [ -v ] [ -V ] [ -o localzone ] [ -l len ]\n\n",
+      basename(argv[0]));
+   fprintf(stderr, "  -r zonefile         the text file with the zone information. Use \"-\" for standard input.\n");
+   fprintf(stderr, "  -d database         the output database filename.\n");
+   fprintf(stderr, "  -v                  verbose mode.\n");
+   fprintf(stderr, "  -V                  show version information, and quit (don't do anything).\n");
+   fprintf(stderr, "  -o localzone        use localzone. Default is 1.\n");
+   fprintf(stderr, "  -l len              len is the length of areacodes.\n\n");
+
 	exit(EXIT_FAILURE);
 }
 
@@ -348,8 +355,9 @@
 
 	if (argc < 2)
 		usage(argv);
-	while ( (c=getopt(argc, argv, "vVr:d:o:l:")) != EOF) {
+	while ( (c=getopt(argc, argv, "hvVr:d:o:l:")) != EOF) {
 		switch (c) {
+         case 'h' : usage(argv);
 			case 'v' : verbose = true; break;
 			case 'V' : printf("%s: V%s Db=%s\n",
 				basename(argv[0]), progversion, dbv); exit(1);
